jobbergate-docs/docs/source/developer_guide/integration_testing.md

```diff
@@ -0,0 +1,533 @@
+# Integration Testing
+
+While conducting integration testing for Jobbergate, it's critical to examine the entire
+cycle of the platform, ranging from the creation of an Application to remote
+Job Submission via the Jobbergate Agent.
+
+To test most of the platforms functionality, the `docker-compose` setup located in the
+[Jobbergate Composed](https://github.com/omnivector-solutions/jobbergate/jobbergate-composed)
+sub-project is sufficient. Begin by referring to the guide in that sub-project's README.
+Pay close attention to the execution of Jobbergate CLI commands as they play a
+significant role in integration testing. For testing you can use the pre-configured user
+credentials:
+
+- **Username:** local-user
+- **Password:** local
+
+Integration testing should cover the following work-flows:
+
+- Logging in through the CLI
+- Creating an Application
+- Querying a single Application
+- Updating an Application
+- Rendering a Job Script from an Application
+- Updating a Job Script
+- Submitting a Job Script
+- Verifying a Job Submission
+- Deleting the Job Submission, Job Script, and Application
+- Logging out through the CLI
+
+
+## Setup
+
+To begin, you will need two separate terminals open. Change directory to the
+`jobbergate-composed` sub-project of the toplevel `jobbergate` folder.
+
+First, you need to start up the Jobbergate platform with docker-compose. In one of your
+terminals, run the following command:
+
+```shell
+docker-compose up --build
+```
+
+Once all the services are started, jump into the prepared `jobbergate-cli` container
+to execute CLI commands. To do so, execute this command in the other terminal you have
+prepared:
+
+```shell
+docker-compose run jobbergate-cli bash
+```
+
+Now you may start executing commands with the Jobbergate CLI.
+
+To assist with some of the commands below, create a `NAME` environment variable that
+will help to identify resources that you create during the process. You should
+set the value based on the current date so that the associated resources are easy to
+identify. Run the following command to set it:
+
+```shell
+export NAME="test--$(whoami)--$(date -I)"
+```
+
+You have now created a test name like `test--tbeck--2023-10-13`.
+
+
+## Logging in through the CLI
+
+The first work-flow you will test covers the auth mechanics of both the CLI and the API.
+
+Run the following command in the Jobbergate CLI:
+
+```shell
+jobbergate login
+```
+
+Next, open the link that is printed out and log in as `local-user` (password "local").
+If asked, grant all of the permissions.
+
+Verify that the CLI reports that the user has been successfully logged in.
+
+At this point, verify that the token that has been retrieved for the user is correct.
+
+Run the following command in the CLI:
+
+```shell
+jobbergate show-token --plain
```

Comment:
```suggestion
jobbergate show-token --decode
```

Decoding from the cli avoids extra steps, like going to jwt.io, but I see `aud` doesn't show there.